Common/Core/RecoDecay.h

```diff
@@ -683,6 +688,7 @@ class RecoDecay
     int indexMother = -1;                  // index of the mother particle
     std::vector<int> arrAllDaughtersIndex; // vector of indices of all daughters of the mother of the first provided daughter
     std::array<int, N> arrDaughtersIndex;  // array of indices of provided daughters
+    int dauCounter = 0;                    // counter particleMother's daughters coming from the decay process
```

Comment:
Why is this needed?

Reply:
Having a counter is needed because the original version of the function was checking that the number of daughters coming from the decay was not larger than the number of expected final daughters, so it was simply checking on the size of the daughters array. Now I have introduced a counter because I still have to check all the daughters, but I should increase the counter only if the production process is a decay.